tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/parallel/RetryAnalyzer.java

```diff
@@ -0,0 +1,40 @@
+package org.alfresco.elasticsearch.parallel;
+
+import java.util.ConcurrentModificationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.IRetryAnalyzer;
+import org.testng.ITestResult;
+
+/**
+ * Sometimes REST tests fail when run in parallel with:
+ * <pre>
+ *     java.lang.IllegalStateException: Invalid use of BasicClientConnManager: connection still allocated.
+ *     Make sure to release the connection before allocating another one.
+ * </pre>
+ * Other times restassured fails with <code>java.util.ConcurrentModificationException</code>.
+ * If we detect a test failed due to these timing issues then we rerun it up to three times.
+ */
+public class RetryAnalyzer implements IRetryAnalyzer
+{
+    private static Logger LOGGER = LoggerFactory.getLogger(RetryAnalyzer.class);
+    private static int RETRY_LIMIT = 3;
+    private int retryNumber = 0;
```

Comment:
I am not sure if `retry` method here can be called from multiple threads but if so, it would be much safer to use `AtomicInteger` instead of `int` as `retryNumber`